### PR TITLE
(RK-386) Remove deprecated `bare` environment type

### PR DESCRIPTION
**Pull request by Magisus**
This environment type was previously renamed to `plain` to better align
with Puppet's terminology. This commit removes the deprecated `bare`
vesrion.

### CODE REVIEW

**Review by bastelfreak — APPROVED** [review]

File: CHANGELOG.mkd [hunk 1]
```diff
@@ -3,6 +3,7 @@ CHANGELOG
 
 Unreleased
 ----------
+- (RK-386) Remove deprecated `bare` environment. [#1235](https://github.com/puppetlabs/r10k/issues/1235)
```
Comment (mwaggett) [inline CHANGELOG.mkd]: (nit)
```suggestion
- (RK-386) Remove deprecated `bare` environment type. [#1235](https://github.com/puppetlabs/r10k/issues/1235)
```

**Review by mwaggett — APPROVED** [review]

**Comment by Magisus** [comment]
Blocked on https://github.com/puppetlabs/r10k/pull/1257. I was going to try to do the mergeup as part of this but couldn't get it to look right.

**Comment by Magisus** [comment]
Okay rebased.

File: CHANGELOG.mkd [hunk 2]
```diff
@@ -3,8 +3,9 @@ CHANGELOG
 
 Unreleased
 ----------
-- (RK-383) Remove deprecated `basedir` method from Puppetfile DSL. Users should use `environment_name` instead. [#1254](https://github.com/puppetlabs/r10k/pull/1254)
 
+- (RK-383) Remove deprecated `basedir` method from Puppetfile DSL. Users should use `environment_name` instead. [#1254](https://github.com/puppetlabs/r10k/pull/1254)
+- (RK-386) Remove deprecated `bare` environment type. [#1235](https://github.com/puppetlabs/r10k/issues/1235)
```
Comment (mwaggett) [inline CHANGELOG.mkd:8]:
```suggestion
- (RK-386) Remove deprecated `bare` environment type. [#1253](https://github.com/puppetlabs/r10k/issues/1253)
```

Comment (Magisus) [reply CHANGELOG.mkd:8]: Good catch 😉

Comment (mwaggett) [reply CHANGELOG.mkd:8]: oh I was wrong, the _issue_ _is_ #1235 lol

Comment (mwaggett) [reply CHANGELOG.mkd:8]: weird coincidence 😅

**Review by mwaggett — APPROVED** [review]

**Pull request merged**